### DISCUSSION TITLE
Return an empty Vec of emergency contacts when emergency access is disabled

### DISCUSSION THREAD

**Discussion by zacknewman** [dani-garcia/vaultwarden#4223] [Ideas]
When emergency access is disabled, an error is returned; however this causes a never-ending spinning "wheel" under "Trusted emergency contacts" and "Designated as emergency contact" when one views the Emergency access information.

I propose instead of erring that an empty `Vec` be returned. Specifically something like

```rust
#[get("/emergency-access/trusted")]
async fn get_contacts(headers: Headers, mut conn: DbConn) -> Json<Value> {
    if !emergency_access_allowed() {
        return Json(json!({
          "Data": Vec::<Value>::new(),
          "Object": "list",
          "ContinuationToken": null
        }));
    }
    ⋮
}
#[get("/emergency-access/granted")]
async fn get_grantees(headers: Headers, mut conn: DbConn) -> Json<Value> {
    if !emergency_access_allowed() {
        return Json(json!({
          "Data": Vec::<Value>::new(),
          "Object": "list",
          "ContinuationToken": null
        }));
    }
    ⋮
}
```

**Comment by stefan0xC** [comment]
Thanks for the suggestion. I've made a PR https://github.com/dani-garcia/vaultwarden/pull/4227 to implement it.

**Comment by Gerardv514** [comment]
@zacknewman nice work with the enhancements/fixing!
